### DISCUSSION TITLE
Time saver. Save compile settings

### DISCUSSION THREAD

**Discussion by dformdotdk** [joomengine/Joomla-Component-Builder#710] [Ideas]
More often than not, we use the same compile settings again and again during the development of a component. 

It would be nice, if JCB could remember the settings and selected component from last compile - at least in the current session. Then we could just click compile component, without having to set the same settings again and again on each compile.

**Comment by Llewellynvdm** [comment]
Okay so here is the settings:
![image](https://user-images.githubusercontent.com/5607939/112230052-dd150200-8c3c-11eb-92df-73011d4e65f7.png)

We have only two that is not having a global setting the `Add to Backup Folder & Sales Server (if set)` and `Add to Repository Folder` both which are set to the default option for development. Since during development you don't want to move to a sales or backup server, and yet you want code to be placed in the repository. The other setting fall back to what you're global in each component. So you can in the component set the default you want during development, and when ready... which is the least compiling moments... then you can manually override those in the compiler... to push the component to the sales server and stuff... not sure why we would need to change this workflow.

But I am open to more perspectives... so explain to me why the above mentioned outcome does not work.

**Reply by dformdotdk** [reply]
I've created a small screen recording to better explain my suggestion: https://youtu.be/8y6tYsZ_3ds

**Reply by dformdotdk** [reply]
As I explain in the video, I had problems getting the "Run expansion" function to work. I got it to work now - just had to follow your fine guide here and add an API User: https://www.youtube.com/watch?v=ozp6m12Fj0o&list=PLQRGFI8XZ_wtGvPQZWBfDzzlERLQgpMRE&index=77 :-) 

There is a great deal of "overlap" between run expansion and my suggestion. But still think it would be nice, if JCB would remember the settings and component selected on the compile view.

**Reply by Llewellynvdm** [reply]
Thanks for the video... here is my little quick video in reply.
https://youtu.be/mwiL4bx0ff0

**Reply by dformdotdk** [reply]
Thanks. Regarding your question about interview, I'd be happy to participate.
